packages/animation/src/withAnimator/withAnimator.ts

```diff
@@ -15,11 +15,11 @@ interface WithAnimatorOutputProps {
   animator?: AnimatorInstanceSettings
 }
 
-function withAnimator (classAnimator?: AnimatorClassSettings) {
+function withAnimator<T extends Component<WithAnimatorInputProps> | FC<WithAnimatorInputProps>> (classAnimator?: AnimatorClassSettings) {
```

Comment:
I'm unsure if there is some better way for this - ideally we would want any component type, which extends WithAnimatorInputProps.

Reply:
The `React.ComponentType` is an alias for `FC` and `ClassComponent` so I think it can be used here.

Reply:
Do you think this change can be also applied for the `extendAnimator()` HOC too?
See here: https://github.com/arwes/arwes/blob/next/packages/animation/src/extendAnimator/extendAnimator.ts#L12-L36
And test case here: https://github.com/arwes/arwes/blob/next/packages/animation/src/extendAnimator/extendAnimator.test.tsx#L80-L94

Reply:
I made the test and it didn't work. Never mind.